src/EventListener/WriteListener.php

```diff
@@ -65,7 +65,7 @@ public function onKernelView(GetResponseForControllerResultEvent $event)
                 // Controller result must be immutable for _api_write_item_iri
                 // if it's class changed compared to the base class let's avoid calling the IriConverter
                 // especially that the Output class could be a DTO that's not referencing any route
-                if (null !== $this->iriConverter && (false !== $attributes['output_class'] ?? null) && \get_class($controllerResult) === \get_class($event->getControllerResult())) {
+                if (null !== $this->iriConverter && (false !== $attributes['output_class'] ?? null) && $attributes['output_class'] === ($class = \get_class($controllerResult)) && $class === \get_class($event->getControllerResult())) {
```

Comment:
Don't use output_class then, just add a check that the controllerResult class is the same as the resource_class I think that it'll be sufficient.

Reply:
@soyuka sure I'll try and update, my concern was, is there a chance that we could be dealing with doctrine proxies here sometime ?

Reply:
No problem with this I think.